src/main/java/com/Elessar/app/server/RegisterHandler.java

```diff
@@ -35,12 +43,23 @@ public void handle(HttpExchange he) throws IOException {
         try (final InputStream is = he.getRequestBody()) {
             final RegistrationRequest regRequest = RegistrationRequest.parseFrom(is);
             final RegistrationResponse.Builder regResponse = RegistrationResponse.newBuilder();
-
-            if (userData.containsKey(regRequest.getName())) {
-                regResponse.setSuccess(false).setFailReason("User Name " + regRequest.getName() + " Exists !");
+            final String userName = regRequest.getName();
+            final Map<String, String> filters = new HashMap<>();
+            filters.put("name", regRequest.getName());
+            Iterator cursor = users.find(filters).iterator();    // how to handle possible os leak?
```

Comment:
Similar to logon and logoff, here we are doing 2 DB operations, one query and one update (if not found). 

Many we should just call insert() and see if mongoDB successfully insert the user. If succeeded, then we are good. If a user exists before then an exception would thrown.
Notice that for this to work, we would need to have a unique index on "user name" in Mongo's user collection